app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -129,11 +129,21 @@ class SmartTransactionHook {
       'ApprovalController:startFlow',
     );
     this.#approvalFlowId = approvalFlowId;
+    let getFeesResponse;
     try {
-      const getFeesResponse = await this.#smartTransactionsController.getFees(
+      getFeesResponse = await this.#smartTransactionsController.getFees(
         { ...this.#txParams, chainId: this.#chainId },
         undefined,
       );
+    } catch (error) {
+      log.error(
+        'Error in smart transaction publish hook, falling back to regular transaction submission',
+        error,
+      );
+      this.#onApproveOrReject();
+      return useRegularTransactionSubmit; // Fallback to regular transaction submission.
```

Comment:
Could we document why we fallback to a regular transaction submission here, but not further down the function?

Reply:
It's because if the `/submitTransactions` call fails, we cannot be 100% sure that it didn't already submit a transaction down the line.